thy/Simple_Firewall/SimpleFw_Semantics.thy

```diff
@@ -367,10 +367,11 @@ Additionally, prefixes should be zero on bits beyond the prefix length.
 \<close>
 
 (*TODO: delete or move*)
-definition "split f a \<equiv> (case a of (x,y) \<Rightarrow> f x y)"
-lemma "curry \<circ> split = id" by(simp add: fun_eq_iff split_def)
+definition "uncurry f \<equiv> (\<lambda>a. case a of (x,y) \<Rightarrow> f x y)"
+declare uncurry_def[simp]
```

Comment:
is this really necessary? We probably only want `uncurry f (a,b) = f (a,b)` as `[simp]`. 
I guess `uncurry f x` as `[simp]` (where `x` is not split) just makes things more complicated.